src/test/java/edu/cornell/kfs/rass/batch/service/impl/RassSortServiceImplTest.java

```diff
@@ -0,0 +1,158 @@
+    private static final String LEFT_OVER_AGENCY_TEST_FILE_NAME = RassXmlDocumentWrapperMarshalTest.RASS_EXAMPLE_FILE_BASE_PATH + "rass-agency-left-over-test.xml";
+    private static final String SINGLE_AWARD_FILE = RassXmlDocumentWrapperMarshalTest.RASS_EXAMPLE_FILE_BASE_PATH + "rass_single_award_only.xml";
+    
+    private static final Logger LOG = LogManager.getLogger(RassSortServiceImplTest.class);
+    private CUMarshalService cuMarshalService;
+    private RassSortServiceImpl rassSortServiceImpl;
+
+    @Before
+    public void setUp() throws Exception {
+        Configurator.setLevel(RassSortServiceImpl.class.getName(), Level.DEBUG);
+        cuMarshalService = new CUMarshalServiceImpl();
+        rassSortServiceImpl = new RassSortServiceImpl();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        cuMarshalService = null;
+        rassSortServiceImpl = null;
+    }
+
+    @Test
+    public void testFullExtractBeforeSort() throws JAXBException {
+        File xmlFile = new File(FULL_EXTRACT_FILE_NAME);
+        RassXmlDocumentWrapper rassXmlDocumentWrapper = cuMarshalService.unmarshalFile(xmlFile, RassXmlDocumentWrapper.class);
+        
+        TestResults results = checkReportsToAgencyBeforeChildAgencies(rassXmlDocumentWrapper.getAgencies());
+        LOG.info("testFullExtractBeforeSort, processed: " + results.processedAgencies);
+        LOG.info("testFullExtractBeforeSort, failed: " + results.failedAgencies);
+        assertTrue("There shoulld be some agencies before their reports to agency before the sort", results.failedAgencies.size() > 0);
+    }
+    
+    @Test
+    public void testFullExtractAfterSort() throws JAXBException {
+        File xmlFile = new File(FULL_EXTRACT_FILE_NAME);
+        RassXmlDocumentWrapper rassXmlDocumentWrapper = cuMarshalService.unmarshalFile(xmlFile, RassXmlDocumentWrapper.class);
+        
+        int preSortCount = rassXmlDocumentWrapper.getAgencies().size();
+        List<RassXmlAgencyEntry> sortedAgencyEntries = rassSortServiceImpl.sortRassXmlAgencyEntriesForUpdate(rassXmlDocumentWrapper.getAgencies());
+        int postSortCount = sortedAgencyEntries.size();
+        
+        validateSortCount(preSortCount, postSortCount);
+        
+        TestResults results = checkReportsToAgencyBeforeChildAgencies(sortedAgencyEntries);
+        LOG.info("testFullExtractAfterSort, processed: " + results.processedAgencies);
+        LOG.info("testFullExtractAfterSort, failed: " + results.failedAgencies);
+        assertTrue("There shoulld be no agencies before their reports to agency after the sort", results.failedAgencies.size() == 0);
+    }
+
+    private void validateSortCount(int preSortCount, int postSortCount) {
+        assertEquals("The sort should return the same nnumber of agencies as was passed in", preSortCount, postSortCount);
```

Comment:
having the assertEquals buried in a validateSortCount method makes this more confusing. I think you should remove the validateSortCount and just call the assertEquals. That will make the code more clear.

Reply:
A different name for the method sounds good

Reply:
New name looks better but I wonder if you meant that to be assertListCountsMatch instead of assortListCountsMatch?

Reply:
There is also an extra n in "nnumber"